src/main/java/com/globalitgeeks/examninja/service/ExamResultService.java

```diff
@@ -116,7 +116,9 @@ private Map<Long, String> filterStudentTestAnswers(Map<String, String> allAnswer
             Long questionId = Long.parseLong(keyParts[2]);
 
             if (storedUserId.equals(userId) && storedTestId.equals(testId)) {
-                studentTestAnswers.put(questionId, value);
+                String selectedAnswer = value.get("label");
```

Comment:
extract this label in a constant.. and give a meaningful name so that code becomes easier to understand..
it should be private static final String SOME_MEANINGFUL_NAME = "label"; choose appropriate name as per the user story..

Reply:
private static final String SELECTEDOPTIONFROMMAP = "label"; 
String selectedAnswer = value.get(SELECTEDOPTIONFROMMAP);